wgsl/extract-grammar.py

```diff
@@ -502,17 +743,196 @@ def read_spec(options):
                             last_value = None
                             result[scanner_span.name()][last_key] = []
                     else:
-                        # It's example text
+                        # A token or example
                         if scanner_parse[1] != None:
                             last_value = scanner_parse[1]
-                            result[scanner_span.name()][last_key].append(
-                                last_value)
-                    scanner_i += scanner_parse[-1]  # Advance line index
+                            if scanner_span.name() == scanner_example.name():
+                                result[scanner_span.name()][last_key].append(
+                                    last_value)
+                            if scanner_span.name() == scanner_token.name():
+                                result[scanner_span.name()][last_key] = last_value
+                    scanner_i += scanner_parse[-1]  # Advance line index 
+        for j in scanner_spans:
+            if scanner_span == j:
+                # Check if we should stop using this scanner.
+                scanner_end = j.end(scanner_lines, scanner_i)
+                if scanner_end[0]:
+                    # Yes, stop using this scanner.
+                    scanner_span = None
+                    scanner_record = False
+                    last_key = None
+                    last_value = None
+                    scanner_i += scanner_end[-1]
         scanner_i += 1
 
-    result[scanner_rule.name()]["_comment"] = [["`/\\/\\/.*/`"]]
+    bnfdialect_data = ""
+
+    # Read the contents of the bnfdialect file
+    with open(options.syntax_filename, "r") as file:
+        bnfdialect_data = file.read()
+
+    syntax_dict = {}
+    current_rule = None
+    for line in bnfdialect_data.splitlines():
+        comment = ""
+        if "//" in line:
+            line, comment = line.split("//", 1)
+        line = line.strip() # Either blankspace is insignificant
+        comment = comment.rstrip() # Right blankspace is insignificant
```

Comment:
We don't use comment after this, so we can skip this step.

Reply:
When we want to emit comments to HTML, we'll need to refer. Can we preserve?